src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -481,9 +482,11 @@ protected function initializeContainer()
             });
         }
 
+        $buildParametersPass = new RemoveBuildParametersPass();
```

Comment:
let's make this a private property instead, to not have to add the argument on buildContainer?

Reply:
another idea: move adding that pass to ContainerBuilder, and expose the result as ContainerBuilder::getRemovedParameters(), which would also return the values of removed parameters to allow dumpContainer to work?

Reply:
Sorry, another idea, simpler one  :)
Let's just NOT try to keep the list of removed parameters. In error paths, let's use this instead:
```php
if ('.' === ($name[0] ?? '')) {
    throw new LogicException(sprintf('Parameter "%s" not found. It was probably deleted during the compilation of the container.', $name));
}
```

Reply:
And even better: move this in ParamterNotFoundException?

Reply:
Done.